### PR TITLE
Replace `querystring_parser.parse` with `flask.Request.args` in artifact handlers

### PR DESCRIPTION
**Pull request by harupy**
### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

In https://github.com/conda-forge/mlflow-feedstock/pull/187, `querystring_parser` was blocking us from uploading mlflow 2.15.0 to conda-forge. This package hasn't been updated for 3 years (https://github.com/bernii/querystring-parser). I think we should remove it before it bites us again. This PR replaces `querystring_parser.parse` calls in the artifact handlers.

There is one more `querystring_parser.parse` here, I think this is also replaceable but needs more verification. I'll fix this in another PR.

https://github.com/mlflow/mlflow/blob/72ba50b265e05e5c64d9b08f5277732993918fdf/mlflow/server/handlers.py#L479-L485

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

Checked the pages on MLflow UI that use the updated artifact handlers, work fine.

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/server/handlers.py [hunk 1]
```diff
@@ -611,12 +611,8 @@ def wrapper(*args, **kwargs):
...
 @catch_mlflow_exception
 def get_artifact_handler():
-    from querystring_parser import parser
-
-    query_string = request.query_string.decode("utf-8")
-    request_dict = parser.parse(query_string, normalized=True)
-    run_id = request_dict.get("run_id") or request_dict.get("run_uuid")
-    path = request_dict["path"]
+    run_id = request.args.get("run_id") or request.args.get("run_uuid")
+    path = request.args["path"]
```
Comment (harupy) [inline mlflow/server/handlers.py:615]: The query string for `get-artifact` looks like this:

```
http://127.0.0.1:7000/get-artifact?path=model%2FMLmodel&run_uuid=60549ecdac2d4b1391001fc15faede47
```

It's a simple query string.

Comment (harupy) [reply mlflow/server/handlers.py:615]: I manually confirmed UI works fine with this change.

image

Comment (harupy) [reply mlflow/server/handlers.py:615]: The same comment on other two handlers.

**Review by BenWilson2 — APPROVED** [review]
werkzeug in flask's arg parser should decode to utf-8 natively, making this a solid simplification change with no forseeable edge case impacts, as well as eliminating any extraneous parsing that querystring_parser was doing. 
LGTM for the stability fix for the conda build process. Thanks for the adjustment @harupy !

**Pull request merged**